### ISSUE TITLE
Disable HairpinMode, enable PromiscMode as workaround for outstanding docker/moby issue

### ISSUE THREAD

**Issue by leodotcloud** [rancher/rancher#9090]
There is a long outstanding [bug](https://github.com/moby/moby/issues/5618) which can cause docker daemon to hang and show a variety of problematic symptoms.

The community has identified a workaround to disable hairpinnat (see [here](https://github.com/moby/moby/issues/5618#issuecomment-280105097)), but this would cause regressions for containers using hairpin mode. Enabling promiscuous mode on the bridge avoids these regressions (see [here](https://github.com/kubernetes/kubernetes/issues/20096#issuecomment-183770382)).

The upstream CNI is also planning to provide this option: https://github.com/containernetworking/cni/pull/441.

**Comment by leodotcloud** [comment]
Fixed in Network Services v0.2.2, network-manager:v0.7.2

**Comment by sangeethah** [comment]
Tested with Rancher server version - v1.6.3-rc1 using Network Services v0.2.2 and  network-manager:v0.7.3.

promiscuous mode is now enabled on docker0.

```
sangeethahariharan1@sangee163rc1final-10acre-1:~$ ip addr show docker0
3: docker0: <BROADCAST,MULTICAST,PROMISC,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 02:42:54:60:6f:94 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 scope global docker0
       valid_lft forever preferred_lft forever
    inet 10.42.0.1/16 scope global docker0
       valid_lft forever preferred_lft forever
    inet6 fe80::42:54ff:fe60:6f94/64 scope link 
       valid_lft forever preferred_lft forever
sangeethahariharan1@sangee163rc1final-10acre-1:~$ 
```
